connector_node/risingwave-source-test/src/test/java/com/risingwave/connector/SourceTestClient.java

```diff
@@ -0,0 +1,160 @@
+package com.risingwave.connector;
+
+import static org.assertj.core.api.Assertions.fail;
+
+import com.risingwave.proto.ConnectorServiceGrpc;
+import com.risingwave.proto.ConnectorServiceProto;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import io.grpc.Channel;
+import io.grpc.StatusRuntimeException;
+import java.io.*;
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.Random;
+import java.util.UUID;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+public class SourceTestClient {
+    static final Logger LOG = LoggerFactory.getLogger(SourceTestClient.class.getName());
+    private final ConnectorServiceGrpc.ConnectorServiceBlockingStub blockingStub;
+
+    public Properties sqlStmts = new Properties();
+
+    public SourceTestClient(Channel channel) {
+        blockingStub =
+                ConnectorServiceGrpc.newBlockingStub(channel).withMaxInboundMessageSize(16777216);
+        try (InputStream input =
+                getClass().getClassLoader().getResourceAsStream("stored_queries.properties")) {
+            sqlStmts.load(input);
+        } catch (IOException e) {
+            fail("failed to load sql statements", e);
+        }
+    }
+
+    protected static Connection connect(JdbcDatabaseContainer<?> container) {
+        Connection connection = null;
+        try {
+            connection = getDataSource(container).getConnection();
+        } catch (SQLException e) {
+            fail("SQL Exception: {}", e);
+        }
+        return connection;
+    }
+
+    protected static ResultSet performQuery(Connection connection, String sql) {
+        ResultSet resultSet = null;
+        try {
+            Statement statement = connection.createStatement();
+            if (statement.execute(sql)) {
+                resultSet = statement.getResultSet();
+                resultSet.next();
+            } else {
+                LOG.info("updated: " + statement.getUpdateCount());
+            }
+        } catch (SQLException e) {
+            LOG.warn("SQL Exception: {}", e.getMessage());
+        }
+        return resultSet;
+    }
+
+    protected static DataSource getDataSource(JdbcDatabaseContainer<?> container) {
```

Comment:
It is definitely unnecessary to build a brand new DataSource each time for every connection. You should put it in the class field.